files/scripts/update-cluster.sh

```diff
@@ -0,0 +1,137 @@
+      echo "Cluster does not exist"
+    else
+      echo "Cluster patch failed, received response:"
+      cat patchResponse.json
+      echo
+      exit 1
+    fi
+}
+
+function removeNodes {
+  local expectedNodes=$1
+  local authToken=$2
+  local namespace=$3
+  local currentNodes=$(getNodeCountInCurrentCluster "$authToken")
+  local nodes=""
+# if there is no cluster or current nodes are less or equal to expected so no need to remove more, exit
+  if [ "$currentNodes" -lt 2 ] || [ "$currentNodes" -le "$expectedNodes" ]; then
+    echo "No scaling required"
+    exit 0
+  fi
+# if there is a cluster and we wanna scale to 1 node, delete it (we would have exit on the last if in case on no cluster)
+  if [ "$expectedNodes" -lt 2 ]; then
+    echo "Deleting cluster"
+    deleteCluster "$authToken"
+    exit 0
+  fi
+
+  for ((i = expectedNodes; i < currentNodes; i++)) do
+    nodes=${nodes}\"graphdb-node-$i.graphdb-node.${namespace}.svc.cluster.local:7300\"
+    if [ $i -lt $(expr $currentNodes - 1) ]; then
+      nodes=${nodes}\,
+    fi
+  done
+  nodes=\{\"nodes\":\[${nodes}\]\}
+  waitService "http://graphdb-cluster-proxy:7200/proxy/ready" "$authToken"
+  curl -o clusterRemove.json -isSL -m 15 -X DELETE --header 'Content-Type: application/json' --header 'Accept: application/json' --header "Authorization: Basic ${authToken}" -d "${nodes}"  'http://graphdb-cluster-proxy:7200/rest/cluster/config/node'
+  if grep -q 'HTTP/1.1 200' "clusterRemove.json"; then
+    echo "Scaling down successful."
+  else
+    echo "Issue scaling down:"
+    cat clusterRemove.json
+    echo
+    exit 1
+  fi
+}
+
+function addNodes {
+  local expectedNodes=$1
+  local authToken=$2
+  local namespace=$3
+  local timeout=$4
+  local currentNodes=$(getNodeCountInCurrentCluster "$authToken")
+  local nodes=""
+# if there is no cluster or current nodes are more or equal to expected so no need to add more, exit
+  if [ "$currentNodes" -lt 2 ] || [ "$currentNodes" -ge "$expectedNodes" ]; then
+    echo "No scaling required"
+    exit 0
+  fi
+  for ((i = currentNodes; i < expectedNodes; i++)) do
+    nodes=${nodes}\"graphdb-node-$i.graphdb-node.${namespace}.svc.cluster.local:7300\"
+    if [ $i -lt $(expr $expectedNodes - 1) ]; then
+      nodes=${nodes}\,
+    fi
+  done
+  nodes=\{\"nodes\":\[${nodes}\]\}
+  waitService "http://graphdb-cluster-proxy:7200/proxy/ready" "$authToken"
+  curl -o clusterAdd.json -isSL -m ${timeout} -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' --header "Authorization: Basic ${authToken}" -d "${nodes}"  'http://graphdb-cluster-proxy:7200/rest/cluster/config/node'
+  if grep -q 'HTTP/1.1 200' "clusterAdd.json"; then
+    echo "Scaling successful."
+  elif grep -q 'Mismatching fingerprints\|HTTP/1.1 412' "clusterAdd.json"; then
+    echo "Issue scaling:"
+    cat clusterAdd.json
+    echo
+    echo "Manual clear of the mismatched repositories will be required to add the node"
+    exit 0
```

Comment:
that's misleading, if I'm scaling up and the script exits without an issue I won't really go to check the logs ... i'll see that the scaling failed through graphdb (probably) and will go back to see the logs (which will be gone by that moment)

Reply:
The reason for the exit 0 so it doesn't go on retrying as this is an unrecoverable error where more retries won't fix it, but you are absolutely right!

Reply:
Would a hacky patch of the job to suspend be preferable or leave it retry to until upgrade fail?

Reply:
Currently set to exit 1